File: library/src/main/java/com/shxhzhxx/library/UrlLoaderCache.java
package com.shxhzhxx.library;

import android.support.annotation.IntRange;
import android.support.annotation.NonNull;

import java.io.File;

public class UrlLoaderCache extends DiskLruCache {
    private final String mSuffixData = "6aef4_d";
    private final String mSuffixHeader = "6aef4_h";

    public UrlLoaderCache(@NonNull File cachePath, @IntRange(from = 1) int maxSize) {
        super(cachePath, maxSize);
    }

    @Override
    public boolean accept(File dir, @NonNull String name) {
        return name.endsWith("." + mSuffixData);
    }

    @Override
    protected int sizeOf(@NonNull File file) {
        return (int) (file.length() + findHeaderCache(file).length());
    }

    @Override
    protected void onDelete(Info info) {
        findHeaderCache(info.file).delete();
        info.file.delete();
    }

    public File getHeaderCache(String url) {
        return getFile(url, mSuffixHeader);
    }

    public File getDataCache(String url) {
        return getFile(url, mSuffixData);
    }

    public boolean clearCache(String url) {
        return remove(getDataCache(url).getName()) != null;
    }

    private File findHeaderCache(File dataCache) {
        String absolutePath = dataCache.getAbsolutePath();
        return new File(absolutePath.substring(0, absolutePath.length() - mSuffixData.length()) + mSuffixHeader);
    }

}

File: library/src/main/java/com/shxhzhxx/urlloader/UrlLoaderCache.kt
package com.shxhzhxx.urlloader

import androidx.annotation.IntRange
import java.io.File

class UrlLoaderCache(cachePath: File, @IntRange(from = 1) maxSize: Int, seed: String = cachePath.absolutePath) : FileLruCache(cachePath, maxSize) {
    private val suffixData = md5("UrlLoaderCache.suffixData$seed").takeLast(8)
    private val suffixHeader = md5("UrlLoaderCache.suffixHeader$seed").takeLast(8)
    override fun accept(dir: File?, name: String?) = name != null && name.endsWith(suffixData)
    override fun sizeOf(file: File) = (file.length() + findHeaderCache(file).length()).toInt()
    override fun onDelete(info: FileInfo) {
        findHeaderCache(info.file).delete()
        info.file.delete()
    }

    fun getHeaderCache(url: String) = getFile(url, suffixHeader)
    fun getDataCache(url: String) = getFile(url, suffixData)
    fun clearCache(url: String): Boolean {
        val cache = getDataCache(url)
        val bool = findHeaderCache(cache).delete()
        return remove(cache.name) != null || bool
    }

    fun findHeaderCache(dataCache: File) = File(dataCache.absolutePath.dropLast(suffixData.length) + suffixHeader)
}